File: app/lib/models/article.dart
class Article {
  final int id;
  final String title;
  final String content;
  final String contentHtml;
  final Map<String, dynamic>? user;
  final Map<String, dynamic>? category;
  final int viewCount;
  final int likeCount;
  final int commentCount;
  final String createdAt;

  Article({
    required this.id,
    required this.title,
    required this.content,
    required this.contentHtml,
    this.user,
    this.category,
    required this.viewCount,
    required this.likeCount,
    required this.commentCount,
    required this.createdAt,
  });

  factory Article.fromJson(Map<String, dynamic> json) {
    return Article(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      contentHtml: json['content_html'] ?? json['contentHtml'] ?? '',
      user: json['user'] as Map<String, dynamic>?,
      category: json['category'] as Map<String, dynamic>?,
      viewCount: json['view_count'] ?? json['viewCount'] ?? 0,
      likeCount: json['like_count'] ?? json['likeCount'] ?? 0,
      commentCount: json['comment_count'] ?? json['commentCount'] ?? 0,
      createdAt: json['created_at'] ?? json['createdAt'] ?? '',
    );
  }
}

File: backend/models/article.go
package models

import (
	"time"
)

type Article struct {
	ID            uint      `gorm:"primarykey" json:"id"`
	Title         string    `gorm:"size:200;not null" json:"title"`
	Content       string    `gorm:"type:text" json:"content"`
	ContentHTML   string    `gorm:"type:text" json:"content_html"`
	UserID        uint      `gorm:"index" json:"user_id"`
	User          User      `gorm:"foreignKey:UserID" json:"user"`
	CategoryID    uint      `gorm:"index" json:"category_id"`
	Category      Category  `gorm:"foreignKey:CategoryID" json:"category"`
	LikeCount     int       `gorm:"default:0" json:"like_count"`
	CommentCount  int       `gorm:"default:0" json:"comment_count"`
	FavoriteCount int       `gorm:"default:0" json:"favorite_count"`
	ViewCount     int       `gorm:"default:0" json:"view_count"`
	ShareCount    int       `gorm:"default:0" json:"share_count"`
	CoinCount     int       `gorm:"default:0" json:"coin_count"`       // 投币数
	Status        string    `gorm:"default:published" json:"status"`   // published, draft, deleted, pending
	IsPinned      bool      `gorm:"default:false" json:"is_pinned"`    // 是否置顶
	PinnedAt      *time.Time `json:"pinned_at"`                         // 置顶时间
	VoiceURL      string    `gorm:"size:500" json:"voice_url"`         // 语音文件URL
	IsAnonymous   bool      `gorm:"default:false" json:"is_anonymous"` // 是否匿名发布
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}
